src/raw-message.ts

```diff
@@ -0,0 +1,42 @@
+import { Client, CommandInteraction, BaseGuildTextChannel } from 'discord.js';
+import { SimpleCommand } from './commands.js';
+
+export default {
+	command: new SimpleCommand(
+		'raw',
+		async (client: Client, interaction: CommandInteraction) => {
+			await interaction.deferReply({ ephemeral: true });
+			const link = interaction.options.getString('message', true);
+			const match = link.match(/(?:discord\.com|discord:\/\/-)\/channels\/(?<guildId>\d{18})\/(?<channelId>\d{18})\/(?<messageId>\d{18})/);
+			if (match) {
```

Comment:
To reduce the amount of nesting in the code, I'd prefer if the "happy" path was the least-indented one. So turn this `if` into a negative test and pull the `else`` contents up here with an early return.